src/services/FeedDataCollector.ts

```diff
@@ -0,0 +1,93 @@
+import {inject, injectable} from 'inversify';
+import {Logger} from 'winston';
+import newrelic from 'newrelic';
+
+import {FeedRepository} from '../repositories/FeedRepository';
+import FeedDataProcessor from './FeedDataProcessor';
+import Settings from '../types/Settings';
+import Feeds, {FeedInput} from '../types/Feed';
+import {PriceService} from './PriceService';
+import {DatumService} from './DatumService';
+import CryptoCompareWSInitializer from './CryptoCompareWSInitializer';
+import PolygonIOPriceInitializer from './PolygonIOPriceInitializer';
+
+@injectable()
+export class FeedDataCollector {
+  @inject('Settings') settings!: Settings;
+  @inject('Logger') logger!: Logger;
+  @inject(FeedRepository) feedRepository!: FeedRepository;
+  @inject(FeedDataProcessor) feedDataProcessor!: FeedDataProcessor;
+  @inject(PriceService) priceService!: PriceService;
+  @inject(DatumService) datumService!: DatumService;
+  @inject(CryptoCompareWSInitializer) cryptoCompareWSInitializer!: CryptoCompareWSInitializer;
+  @inject(PolygonIOPriceInitializer) polygonIOPriceInitializer!: PolygonIOPriceInitializer;
+
+  async apply(): Promise<void> {
+    this.initializeWSFetchers();
+
+    const [fcdFeeds, leafFeeds] = await Promise.all([
+      this.feedRepository.getFcdFeeds(),
+      this.feedRepository.getLeafFeeds(),
+    ]);
+
+    if (!Object.keys(fcdFeeds).length && !Object.keys(leafFeeds).length) {
+      this.logger.error('[FeedDataCollector] FCD and L2D feeds are empty');
+      return;
+    }
+
+    const feeds = this.mergeFeedsIntoUniqueHttpInputs(fcdFeeds, leafFeeds);
+    console.log('feeds: ', feeds);
+    const timestamp = Math.floor(Date.now() / 1000);
+    const {data, prices} = await this.feedDataProcessor.apply(timestamp, feeds);
+
+    await Promise.all([this.priceService.savePrices(prices), this.datumService.saveData(data)]);
+  }
+
+  private initializeWSFetchers = (): void => {
+    this.polygonIOPriceInitializer.apply().catch((err: Error) => {
+      newrelic.noticeError(err);
+      this.logger.error(err);
+      process.exit(1);
+    });
+
+    this.cryptoCompareWSInitializer.apply().catch((err: Error) => {
+      newrelic.noticeError(err);
+      this.logger.error(err);
+      process.exit(1);
+    });
+  };
+
+  private mergeFeedsIntoUniqueHttpInputs(...feeds: Feeds[]): Feeds {
+    const uniqueFeeds: Feeds = {};
+
+    feeds.forEach((feed) => {
+      Object.keys(feed).forEach((key: string) => {
+        feed[key].inputs.forEach((input) => {
+          if (this.hasWSFetcher(input)) {
+            return;
+          }
+
+          if (!uniqueFeeds[key]) {
+            uniqueFeeds[key] = {discrepancy: feed[key].discrepancy, precision: feed[key].precision, inputs: []};
+          }
+
+          if (!uniqueFeeds[key].inputs.some((feed) => this.hasSameFetcher(feed, input))) {
```

Comment:
it his really worth doing separate method?

```suggestion
          if (!uniqueFeeds[key].inputs.some((feed) => feed.fetcher.name === input.fetcher.name))) {
```

Reply:
Done